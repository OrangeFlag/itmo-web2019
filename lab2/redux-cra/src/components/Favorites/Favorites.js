import React from 'react'
import {Button} from "../Button/Button";
import Input from "../Input/Input";
import Container from "../Container/Container.style";

export class Favorites extends React.Component {

    constructor(props) {
        super(props);
        this.state = {inputTown: ""};
    }


    handleInputChange = e => {
        this.setState({inputTown: e.target.value})
    };

    handleBtnClick = e => {
        this.props.addFavoriteTown(this.state.inputTown)
    };

    render() {
        return (<Container>
            <p>Избранное</p>
            <Input id="new_town" type="text" name="favorite" placeholder="Добавить новый город" maxLength="100"
                   handleChange={this.handleInputChange}/>
            <Button action={this.handleBtnClick}>+</Button>
        </Container>)
    }
}