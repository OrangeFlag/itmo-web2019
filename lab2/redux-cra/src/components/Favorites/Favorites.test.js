import React from 'react';
import {mount} from 'enzyme';
import {Favorites} from './Favorites'

describe('Favorites test', () => {
    it('should render correctly', () => {
        const component = mount(<Favorites/>);

        expect(component).toMatchSnapshot();
        component.unmount();
    });

    // it('input change should change state', () => {
    //     const component = mount(<Favorites/>);
    //
    //     component.find("input").simulate('change', {target: {value: 'Moscow'}});
    //
    //     expect(component.state().inputTown).toEqual("Moscow");
    //
    //     component.unmount();
    // });

    it('button click should call function with correct arguments', () => {
        const clickFn = jest.fn();
        const component = mount(<Favorites addFavoriteTown={clickFn}/>);

        component.find("input").simulate('change', {target: {value: 'MosCow'}});

        component.find('form').simulate('submit', {
            preventDefault() {
            }
        });

        expect(clickFn).toHaveBeenCalledWith("moscow");
        component.unmount();
    });
});