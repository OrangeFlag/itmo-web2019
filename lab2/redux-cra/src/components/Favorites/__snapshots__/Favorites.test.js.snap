// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites test should render correctly 1`] = `
<Favorites>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "ljZRbJ",
            "rules": Array [
              "
  display: flex;
  flex-flow: row wrap;
  align-items: center;

  width: 76%;

  background-color: #0138a5;
  background-color: ___CSS_1___;

  color: #ffffff;
  color: ___CSS_2___;
  
  padding: 12px 10%;
  padding-top: ___CSS_3___;
  padding-bottom: ___CSS_4___;
  
  box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.30);
  margin: 7px;


  * {
    margin-right: 16px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-htpNat ljZRbJ"
      >
        <p>
          Избранное
        </p>
        <form
          onSubmit={[Function]}
        >
          <Input
            handleChange={[Function]}
            id="new_town"
            maxLength="100"
            name="favorite"
            placeholder="Добавить новый город"
            type="text"
          >
            <styled.input
              defaultValue=""
              handleChange={[Function]}
              id="new_town"
              maxLength="100"
              name="favorite"
              onChange={[Function]}
              placeholder="Добавить новый город"
              type="text"
            >
              <StyledComponent
                defaultValue=""
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "jzaxsF",
                      "rules": Array [
                        "
  background-color: ___CSS_0___;
  border: 1px solid ___CSS_1___;
  border-radius: 4px;
  padding: 12px 15px;
  margin: 8px 0;
  outline: none;
  font-size: '1rem';
  transition: ___CSS_2___;

  &:focus {
    border: 1px solid ___CSS_3___;
    transition: ___CSS_4___;
  }
",
                      ],
                    },
                    "displayName": "styled.input",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "input",
                    "toString": [Function],
                    "usesTheme": false,
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                handleChange={[Function]}
                id="new_town"
                maxLength="100"
                name="favorite"
                onChange={[Function]}
                placeholder="Добавить новый город"
                type="text"
              >
                <input
                  className="sc-bwzfXH jzaxsF"
                  defaultValue=""
                  id="new_town"
                  maxLength="100"
                  name="favorite"
                  onChange={[Function]}
                  placeholder="Добавить новый город"
                  type="text"
                />
              </StyledComponent>
            </styled.input>
          </Input>
          <Button
            type="submit"
          >
            <styled.button
              type="submit"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "JEQYc",
                      "rules": Array [
                        "
    border: 0;
    padding: 13px 16px;
    // height: 50px;
    // line-height: 50px;
    // border-radius: ___CSS_0___;
    // box-shadow: ___CSS_1___;
    outline: none;
    display: inline-block;
    text-align: center;
    text-decoration: none;
    font-size: 15px;
    border-radius: 4px;

    &:hover {
        text-decoration: none;
        background: #cccccc;
        color: ___CSS_3___;
    }
   

    &:focus {
        text-decoration: none;
        outline: none;
    }
",
                      ],
                    },
                    "displayName": "styled.button",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "button",
                    "toString": [Function],
                    "usesTheme": false,
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                type="submit"
              >
                <button
                  className="sc-bdVaJa JEQYc"
                  type="submit"
                >
                  +
                </button>
              </StyledComponent>
            </styled.button>
          </Button>
        </form>
      </div>
    </StyledComponent>
  </styled.div>
</Favorites>
`;
